plans/dht/test/find_providers.go

```diff
@@ -5,106 +5,111 @@ import (
 	"fmt"
 	"time"
 
+	"golang.org/x/sync/errgroup"
+
 	"github.com/ipfs/go-cid"
-	ipfsUtil "github.com/ipfs/go-ipfs-util"
+	u "github.com/ipfs/go-ipfs-util"
+
 	"github.com/ipfs/testground/sdk/runtime"
 	"github.com/ipfs/testground/sdk/sync"
 )
 
-// FindProviders implements the Find Providers Test case
 func FindProviders(runenv *runtime.RunEnv) {
-	// Test Parameters
-	var (
-		timeout     = time.Duration(runenv.IntParamD("timeout_secs", 60)) * time.Second
-		randomWalk  = runenv.BooleanParamD("random_walk", false)
-		bucketSize  = runenv.IntParamD("bucket_size", 20)
-		autoRefresh = runenv.BooleanParamD("auto_refresh", true)
-		// pProviding  = runenv.IntParamD("p_providing", 10)
-		// pResolving  = runenv.IntParamD("p_resolving", 10)
-		// pFailing    = runenv.IntParamD("p_failing", 10)
-	)
-
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	opts := &SetupOpts{
+		Timeout:        time.Duration(runenv.IntParamD("timeout_secs", 60)) * time.Second,
+		RandomWalk:     runenv.BooleanParamD("random_walk", false),
+		NFindPeers:     runenv.IntParamD("n_find_peers", 1),
+		BucketSize:     runenv.IntParamD("bucket_size", 20),
+		AutoRefresh:    runenv.BooleanParamD("auto_refresh", true),
+		NodesProviding: runenv.IntParamD("nodes_providing", 10),
+		RecordCount:    runenv.IntParamD("record_count", 5),
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), opts.Timeout)
 	defer cancel()
 
 	watcher, writer := sync.MustWatcherWriter(runenv)
 	defer watcher.Close()
 	defer writer.Close()
 
-	_, dht, _, err := SetUp(ctx, runenv, timeout, randomWalk, bucketSize, autoRefresh, watcher, writer)
+	_, dht, _, seq, err := Setup(ctx, runenv, watcher, writer, opts)
 	if err != nil {
 		runenv.Abort(err)
 		return
 	}
 
-	defer TearDown(ctx, runenv, watcher, writer)
-
-	/// --- Act I
-
-	// TODO, only `p-providing` of the nodes provide a record and store its key on redis
-
-	var (
-		cidsToPublish     []cid.Cid
-		nCidsToPublish    = 5
-		cidsToNotPublish  []cid.Cid
-		nCidsToNotPublish = 5
```

Comment:
These were used to create Find_Peers calls that were impossible to resolve. You don't want that anymore?